src/include/execution/vm/llvm_engine.h

```diff
@@ -190,6 +193,59 @@ class LLVMEngine {
     std::unique_ptr<TPLMemoryManager> memory_manager_;
     std::unordered_map<std::string, void *> functions_;
   };
+
+  // -------------------------------------------------------
+  // LLVM Engine Settings
+  // -------------------------------------------------------
+
+  /**
+   * Engine-wide settings that apply for the entire process in which the LLVM Engine runs.
+   */
+  class Settings {
+   public:
+    /**
+     * Construct a settings instance from the relevant configuration parameters.
+     * @param bytecode_handlers_path The path to the bytecode handlers bitcode file.
+     */
+    explicit Settings(std::string_view bytecode_handlers_path) : bytecode_handlers_path_{bytecode_handlers_path} {}
+
+    /**
+     * @return The path to the bytecode handlers bitcode file.
+     */
+    const std::string &GetBytecodeHandlersBcPath() const noexcept { return bytecode_handlers_path_; }
+
+   private:
+    const std::string bytecode_handlers_path_;
+  };
+
+  // -------------------------------------------------------
+  // Static Data Members
+  // -------------------------------------------------------
+
+  /**
+   * Process-wide LLVM engine settings.
+   *
+   * TODO(Kyle): I'm not particularly happy with this setup - an inline
```

Comment:
I'm also not a huge fan of this setup. No ideas currently come to mind though.

Reply:
Yea... as the comment says I really tried a number of other approaches, but they all complicated the system so much more than should be necessary for such a small increase in functionality.